Fhi.Smittestopp.Verification.Domain/AnonymousTokens/AnonymousTokenValidationKey.cs

```diff
@@ -1,23 +1,45 @@
-﻿using Org.BouncyCastle.Crypto.Parameters;
+﻿using System;
+using Fhi.Smittestopp.Verification.Domain.Dtos;
+using Org.BouncyCastle.Asn1.X9;
+using Org.BouncyCastle.Crypto.Parameters;
 using Org.BouncyCastle.Utilities.Encoders;
 using Org.BouncyCastle.X509;
 
 namespace Fhi.Smittestopp.Verification.Domain.AnonymousTokens
 {
     public class AnonymousTokenValidationKey
     {
-        public AnonymousTokenValidationKey(string kid, ECPublicKeyParameters publicKey)
+        public AnonymousTokenValidationKey(string kid, string curveName, X9ECParameters ecParameters, ECPublicKeyParameters publicKey)
         {
             Kid = kid;
+            CurveName = curveName;
+            EcParameters = ecParameters;
             PublicKey = publicKey;
         }
 
         public string Kid { get; set; }
+        public string KeyType => "EC";
+        public string CurveName { get; set; }
         public ECPublicKeyParameters PublicKey { get; set; }
+        public X9ECParameters EcParameters { get; set; }
 
         public string GetEncodedKey()
         {
             return Hex.ToHexString(SubjectPublicKeyInfoFactory.CreateSubjectPublicKeyInfo(PublicKey).GetEncoded());
         }
+
+        public AnonymousTokenKey AsKeyDto()
+        {
+            return new AnonymousTokenKey
+            {
+                Kid = Kid,
+                Kty = KeyType,
+                Crv = CurveName,
+                X = Convert.ToBase64String(PublicKey.Q.XCoord.GetEncoded()),
```

Comment:
Maybe better with
X = Convert.ToBase64String(PublicKey.Q.AffineXCoord.ToBigInteger().ToByteArray(),
Y = Convert.ToBase64String(PublicKey.Q.AffineYCoord.ToBigInteger().ToByteArray(),

The Affine prefix normalizes the point first (a somewhat deep mathematical concept, but it gives a canonical representation). ToBigInteger() turns it into a number rather than a field element, and hence easier to use in Curve.CreatePoint(BigInteger x, BigInteger y) later

Reply:
Implemented in https://github.com/folkehelseinstituttet/Fhi.Smittestopp.Verification/pull/29